rnd/autogpt_server/autogpt_server/executor/manager.py

```diff
@@ -225,13 +244,17 @@ def add_execution(self, graph_id: str, data: dict[str, Any]) -> dict:
             if not valid:
                 raise Exception(error)
 
-        graph_exec_id, node_execs = self.run_and_wait(create_graph_execution(
-            graph_id=graph_id,
-            node_ids=[node.id for node in graph.starting_nodes],
-            data=data
-        ))
+        graph_exec_id, node_execs = self.run_and_wait(
+            create_graph_execution(
+                graph_id=graph_id,
+                node_ids=[node.id for node in graph.starting_nodes],
+                data=data,
+            )
+        )
+
+        agent_server: AgentServer = get_service_client(AgentServer)  # type: ignore
```

Comment:
Ditto, agent_server can be shared on @property 

And 

```
            execution_result = get_execution_result(
                node_exec.graph_exec_id, node_exec.node_exec_id
            )
            agent_server.send_execution_update(execution_result)  # type: ignore
```

can be bundled on`execution_update` function (that `self.add_node_execution` calls).